src/plan/mutator_context.rs

```diff
@@ -316,11 +309,6 @@ pub trait MutatorContext<VM: VMBinding>: Send + 'static {
     fn get_tls(&self) -> VMMutatorThread;
     /// Get active barrier trait object
     fn barrier(&mut self) -> &mut dyn Barrier<VM>;
-    /// Force cast the barrier trait object to a concrete implementation.
-    ///
-    /// # Safety
-    /// The safety of this function is ensured by a down-cast check.
-    unsafe fn barrier_impl<B: Barrier<VM>>(&mut self) -> &mut B;
```

Comment:
I agree a binding can implement this method totally on the binding side if they need to. We can remove it.

The comments for write barriers methods mention using a specialized slow-path but they do not specifically mention this method. The comments seem to make sense even if we remove this method from our API. No change is required -- I just mention it in case you would like to improve the comments related to the function.
https://github.com/mmtk/mmtk-core/blob/47562b74c73ec74ab46e9b578fe3919346010b77/src/memory_manager.rs#L225

Reply:
Yes.  The comment mentioned slow path, but it does not mention how to make the "specialized slow-path call".  Calling methods of the `Barrier` trait is only one step of it.  I think it is obvious to use downcast to eliminate a dynamic call for a trait that supports it.